File: server/src/main/java/org/dan/jadalnia/app/user/UserService.java
package org.dan.jadalnia.app.user;

import lombok.RequiredArgsConstructor;
import lombok.experimental.FieldDefaults;
import lombok.extern.slf4j.Slf4j;

import javax.inject.Inject;
import java.util.concurrent.CompletableFuture;

@Slf4j
@FieldDefaults(makeFinal = true)
@RequiredArgsConstructor(onConstructor = @__(@Inject))
public class UserService {
    UserDao userDao;

    public CompletableFuture<UserSession> register(UserRegRequest regRequest) {
        return userDao.register(regRequest.getFestivalId(), regRequest.getUserType(),
                UserState.Approved, regRequest.name, regRequest.getSession());
    }
}

File: server/src/main/java/org/dan/jadalnia/app/user/UserService.kt
package org.dan.jadalnia.app.user

import org.dan.jadalnia.app.festival.pojo.Fid
import javax.inject.Inject
import java.util.concurrent.CompletableFuture

class UserService @Inject constructor(val userDao: UserDao) {
  fun register(regRequest: UserRegRequest): CompletableFuture<UserSession> {
    return userDao.register(regRequest.festivalId, regRequest.userType,
        UserState.Approved, regRequest.name, regRequest.session)
  }

  fun listByType(fid: Fid, userType: UserType) =
      userDao.listByType(fid, userType);
}
